src/decoder/stream.rs

```diff
@@ -856,10 +856,18 @@ impl StreamingDecoder {
                 Ok(Decoded::Nothing)
             }
             ColorType::Indexed => {
-                // FIXME: what's going on here??
-                let _ = info.palette.as_ref().ok_or_else(|| {
-                    DecodingError::Format(FormatErrorInner::AfterPlte { kind: chunk::tRNS }.into())
-                });
+                // The transparency chunk must be after the palette chunk and
+                // before the data chunk.
+                if info.palette.is_none() {
+                    return Err(DecodingError::Format(
+                        FormatErrorInner::AfterPlte { kind: chunk::tRNS }.into(),
+                    ));
+                } else if self.have_idat {
+                    return Err(DecodingError::Format(
+                        FormatErrorInner::OutsidePlteIdat { kind: chunk::tRNS }.into(),
+                    ));
```

Comment:
There's a `FIXME` in the variant documentation about this variant not being used and wondering where it _should_ be used. Can you remove it?

Reply:
Good catch! I've amended the pull request to remove the `FIXME` in the variant documentation. Thanks!